302/share tab layout

louisewang1 opened the pull request:
close #302 
close #268

pyphilia commented on src/components/item/sharing/VisibilitySelect.js:
```diff
@@ -58,13 +59,17 @@ const VisibilitySelect = ({ item, edit }) => {
   const [itemTagValue, setItemTagValue] = useState(false);
   const [tagValue, setTagValue] = useState(false);
   const [isDisabled, setIsDisabled] = useState(false);
+  const [open, setOpen] = useState(false);
```
`isPublishedSelected`?

louisewang1 commented:
I added the download button in this branch, so I requested a review again for the new feature I've implemented @pyphilia I decided to put the download Icon outside the sandwich menu, mainly because I made it an Icon Button, but in the menu, all buttons are text ones. Also in explorer, we use the Icon Button, so I try to unify them, and both can use the component from @graasp/ui. I also temporarily disabled a few tests related to publish item. I will work on the tests probably next week,

pyphilia reviewed (changes requested):
Big changes! Keep in mind to split as much as possible the process logic and the UI logic.
